include/effects.h

```diff
@@ -0,0 +1,151 @@
+#define EFFECT_STRIP_SPLASH_LOGO                        38
+
+// Matrix effects
+#define EFFECT_MATRIX_ALIEN_TEXT                       101
+#define EFFECT_MATRIX_BOUNCE                           102
+#define EFFECT_MATRIX_CIRCUIT                          103
+#define EFFECT_MATRIX_CLOCK                            104
+#define EFFECT_MATRIX_CUBE                             105
+#define EFFECT_MATRIX_FLOW_FIELD                       106
+#define EFFECT_MATRIX_LIFE                             107
+#define EFFECT_MATRIX_MANDALA                          108
+#define EFFECT_MATRIX_SUNBURST                         109
+#define EFFECT_MATRIX_ROSE                             110
+#define EFFECT_MATRIX_PINWHEEL                         111
+#define EFFECT_MATRIX_INFINITY                         112
+#define EFFECT_MATRIX_MUNCH                            113
+#define EFFECT_MATRIX_CURTAIN                          114
+#define EFFECT_MATRIX_GRID_LIGHTS                      115
+#define EFFECT_MATRIX_PALETTE_SMEAR                    116
+#define EFFECT_MATRIX_RAINBOW_FLAG                     117
+#define EFFECT_MATRIX_PONG_CLOCK                       118
+#define EFFECT_MATRIX_PULSE                            119
+#define EFFECT_MATRIX_PULSAR                           120
+#define EFFECT_MATRIX_QR                               121
+#define EFFECT_MATRIX_RADAR                            122
+#define EFFECT_MATRIX_SERENDIPITY                      123
+#define EFFECT_MATRIX_SPARK                            124
+#define EFFECT_MATRIX_SPIN                             125
+#define EFFECT_MATRIX_SPIRO                            126
+#define EFFECT_MATRIX_SUBSCRIBERS                      127
+#define EFFECT_MATRIX_SWIRL                            128
+#define EFFECT_MATRIX_WAVE                             129
+#define EFFECT_MATRIX_WEATHER                          130
+#define EFFECT_MATRIX_INSULATOR_SPECTRUM               131
+#define EFFECT_MATRIX_SPECTRUM_ANALYZER                132
+#define EFFECT_MATRIX_WAVEFORM                         133
+#define EFFECT_MATRIX_GHOST_WAVE                       134
+
+// Starry Night star variations
+#define EFFECT_STAR                                      1
+#define EFFECT_STAR_RANDOM_PALETTE_COLOR                 2
+#define EFFECT_STAR_LONG_LIFE_SPARKLE                    3
+#define EFFECT_STAR_COLOR                                4
+#define EFFECT_STAR_MUSIC                                5
+#define EFFECT_STAR_MUSIC_PULSE                          6
+#define EFFECT_STAR_QUIET                                7
+#define EFFECT_STAR_BUBBLY                               8
+#define EFFECT_STAR_FLASH                                9
+#define EFFECT_STAR_COLOR_CYCLE                         10
+#define EFFECT_STAR_MULTI_COLOR                         11
+#define EFFECT_STAR_CHRISTMAS                           12
+#define EFFECT_STAR_HOT_WHITE                           13
+
+// Some common JSON properties to prevent typos
```

Comment:
Is there real value in keep these to 3 chars, or would longer more descriptive names make the JSON easier to read?

Reply:
Three premises for my implementation of this are:
1. Prioritizing minimum use of memory (and storage)
2. Each serializable class decides for itself how it serializes itself into a JSON object
3. The JSON that's saved is really only for "internal use" by the program logic itself, so it does not have to be that human-friendly.

The first premise calls for short property names. The second means that each serializable class only needs to make sense of the JSON property names it uses for its own JSON objects. As the number of properties per class is pretty small, I think the shorter property names are not too confusing and thus defendable.

Of course, this does change if you want to optimize the JSON for consumption by humans, thereby effectively dropping premise 3. 

It's a choice, and we can choose either way. I guess it depends on what use (and composition) of the JSON you envisage. :)